File: examples/vb.net/ConsoleApp.vb
'''
 '  Product:        Tolk
 '  File:           ConsoleApp.vb
 '  Description:    VB.NET console application example.
 '  Copyright:      (c) 2014, Davy Kager <mail@davykager.nl>
 '  License:        LGPLv3
 '  Depends:        Tolk.dll, TolkDotNet.dll
 ''

'' See README.md for general information.
'' See Tolk.h for full documentation of all functions.
'' Note that for .NET the functions are provided as static functions of class Tolk,
'' and that the prefix Tolk_ has been removed from the names.

Imports DavyKager

Module ConsoleApp
  Public Sub Main()
    Console.WriteLine("Tolk -- VB.NET Console App Example")

    Console.WriteLine("Initializing Tolk...")
    Tolk.Load()

    Console.WriteLine("Querying for the active screen reader driver...")
    Dim name As String = Tolk.DetectScreenReader()
    If name IsNot Nothing Then
      Console.WriteLine("The active screen reader driver is: {0}", name)
    Else
      Console.WriteLine("None of the supported screen readers is running")
    End If

    If Tolk.HasSpeech() Then
      Console.WriteLine("This screen reader driver supports speech")
    End If
    If Tolk.HasBraille() Then
      Console.WriteLine("This screen reader driver supports braille")
    End If

    Console.WriteLine("Let's output some text...")
    If Not Tolk.Output("Hello, World!") Then
      Console.WriteLine("Failed to output text")
    End If

    Console.WriteLine("Finalizing Tolk...")
    Tolk.Unload()

    Console.WriteLine("Done!")
  End Sub
End Module

File: contrib/oxygene/ConsoleApp.pas
namespace;

interface

uses
  System,
  DavyKager;

type
  ConsoleApp = public class
  public
    class method Main;
  end;

implementation

class method ConsoleApp.Main;
begin
  Console.WriteLine('Tolk -- C# Console App Example');
  Console.WriteLine('Initializing Tolk...');
  Tolk.Load();
  Console.WriteLine('Querying for the active screen reader driver...');
  var name: String := Tolk.DetectScreenReader();
  if assigned(name) then begin
    Console.WriteLine('The active screen reader driver is: {0}', name);
  end
  else begin
    Console.WriteLine('None of the supported screen readers is running');
  end;
  if Tolk.HasSpeech() then begin
    Console.WriteLine('This screen reader driver supports speech');
  end;
  if Tolk.HasBraille() then begin
    Console.WriteLine('This screen reader driver supports braille');
  end;
  Console.WriteLine('Let''s output some text...');
  if not Tolk.Output('Hello, World!') then begin
    Console.WriteLine('Failed to output text');
  end;
  Console.WriteLine('Finalizing Tolk...');
  Tolk.Unload();
  Console.WriteLine('Done!');
end;

end.
